incubator/hnc/hack/test-issue-797-non-leaf.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+bold=$(tput bold)
+normal=$(tput sgr0)
+
+# Set up namespace structure
+echo "----------------------------------------------------"
+echo "${bold}Creating a race condition of two parents with the same non-leaf subns (anchor)"
+echo "Creating two root namespaces 'p1' and 'p2'...${normal}"
+kubectl create ns p1
+kubectl create ns p2
+sleep 1
+
+echo "${bold}Disabling webhook to generate the race condition...${normal}"
+kubectl delete validatingwebhookconfigurations.admissionregistration.k8s.io hnc-validating-webhook-configuration
+sleep 1
+
+echo "${bold}Creating subns (anchor) 'sub' in both 'p1' and 'p2'...${normal}"
+kubectl hns create sub -n p1
+kubectl hns create sub -n p2
+sleep 1
+
+echo "${bold}Subnamespace 'sub' should be created and have 'p1' as the 'subnamespaceOf' annoation value:${normal}"
+kubectl get ns sub -o yaml
+
+echo "${bold}subns (anchor) 'sub' in 'p2' should have 'status: conflict' because it's a bad anchor:${normal}"
+kubectl get subns sub -n p2 -o yaml
+
+echo "${bold}Creating subns (anchor) 'sub-sub' in 'sub' to make 'sub' non-leaf...${normal}"
+kubectl hns create sub-sub -n sub
+sleep 1
+
+echo "${bold}Setting allowCascadingDelete on 'sub'...${normal}"
+kubectl hns set sub -a
+
```

Comment:
Perhaps you should do this _after_ trying to delete the bad anchor? The bad anchor should be deleted even if ACD _isn't_ set, right?

Reply:
Right. SG, updated.